### ISSUE TITLE
 About solving：Output size is too small

### ISSUE THREAD

**Issue by ljc19940403** [backseason/PoolNet#53]
hello,I need help with a problem at work recently.When the aspect ratio of the detected picture is larger or smaller,Runtime error： Output size is too small.
Do you have any suggestions for this problem?
Can this kind of image be detected by modifying the network layer？

**Comment by backseason** [comment]
This is caused by the pooling layers in the PPM.
When the input image is too small, in the late layers of the network, the spatial size of the feature maps may be smaller than the kernel size of the pooling layers, which will cause the error that you have mentioned.
This error can be simply fixed by resizing the input images to a bigger size.

**Comment by ljc19940403** [comment]
> This is caused by the pooling layers in the PPM.
> When the input image is too small, in the late layers of the network, the spatial size of the feature maps may be smaller than the kernel size of the pooling layers, which will cause the error that you have mentioned.
> This error can be simply fixed by resizing the input images to a bigger size.

After I tried to fix the padding on some PPM layer, it seemed to work and the mask effect was acceptable

**Comment by lucasdavid** [comment]
I solved this by replacing:
```py
if h <= 100 or w <= 100:
  images = F.interpolate(images, scale_factor=2, mode='bilinear')
```
With:
```py
if h <= 100 or w <= 100:
  ratio = 100 / min(h, w)
  sizes = (int(round(ratio * h)), int(round(ratio * w)))
  x = F.interpolate(x, size=sizes, mode='bicubic')
```
Explanation: there's really no guarantee that $2(h, w) > (100, 100)$. For example, if $(h, w) = (40, 500)$.
Hence I upsample it so its shortest size is equal to 100, and the other is proportionally larger (maintaining the aspect ratio).
